File: 167-TwoSumIiInputArrayIsSorted/167-TwoSumIiInputArrayIsSorted.java
// Last updated: 3/14/2026, 11:50:50 AM
class Solution {
    public int[] twoSum(int[] numbers, int target) {
       int left = 0;
       int right = numbers.length -1;
        int a = 0 ;
        int b = 0;
       while(left < right){
        if(numbers[left] + numbers[right] == target){
            a=left +1;
            b = right+1;
            break;
        }
        else if(numbers[left] + numbers[right] > target){
            right--;
        }
        else{
            left++;
        }
       }
       return new int[]{a,b};
        
    }
}

File: 167-TwoSumIiInputArrayIsSorted/167-TwoSumIiInputArrayIsSorted.py
# Last updated: 3/21/2026, 4:40:43 PM
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers)-1

        while left < right:
            if numbers[left] + numbers[right] == target:
                return [left+1, right+1]
            elif numbers[left] + numbers[right] >target:
                right -=1
            elif numbers[left] + numbers[right] < target:
                left+=1;
    